streams/src/main/java/org/apache/kafka/streams/kstream/internals/BaseJoinProcessorNode.java

```diff
@@ -0,0 +1,109 @@
+
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.streams.kstream.ValueJoiner;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+
+/**
+ * Utility base class containing the common fields between
+ * a Stream-Stream join and a KTable-KTable join
+ */
+class BaseJoinProcessorNode<K, V, V1, V2, VR> extends StreamGraphNode<K, V> {
+
+    private final ProcessorSupplier<K, V1> joinThisProcessSupplier;
+    private final ProcessorSupplier<K, V2> joinOtherProcessSupplier;
+    private final ProcessorSupplier<K, VR> joinMergeProcessor;
+    private final ValueJoiner<? super V, ? super V1, ? extends V2> valueJoiner;
+    private final String joinThisProcessorName;
+    private final String joinOtherProcessorName;
+    private final String joinMergeProcessorName;
+    private final String thisJoinSide;
+    private final String otherJoinSide;
+
+
+    BaseJoinProcessorNode(final String predecessorNodeName,
+                          final String name,
+                          final ValueJoiner<? super V, ? super V1, ? extends V2> valueJoiner,
+                          final ProcessorParameters<K, V1> joinThisProcessorDetails,
+                          final ProcessorParameters<K, V2> joinOtherProcessDetails,
+                          final ProcessorParameters<K, VR> joinMergeProcessorDetails,
+                          final String thisJoinSide,
+                          final String otherJoinSide) {
+
+        super(predecessorNodeName,
+              name,
+              false);
+
+        this.valueJoiner = valueJoiner;
+        this.joinThisProcessSupplier = joinThisProcessorDetails.processorSupplier();
+        this.joinOtherProcessSupplier = joinOtherProcessDetails.processorSupplier();
+        this.joinMergeProcessor = joinMergeProcessorDetails.processorSupplier();
+        this.joinThisProcessorName = joinThisProcessorDetails.name();
+        this.joinOtherProcessorName = joinOtherProcessDetails.name();
+        this.joinMergeProcessorName = joinMergeProcessorDetails.name();
+        this.thisJoinSide = thisJoinSide;
+        this.otherJoinSide = otherJoinSide;
+    }
+
+    ProcessorSupplier<K, V1> joinThisProcessorSupplier() {
+        return joinThisProcessSupplier;
+    }
+
+    ProcessorSupplier<K, V2> joinOtherProcessorSupplier() {
+        return joinOtherProcessSupplier;
+    }
+
+    ProcessorSupplier<K, VR> joinMergeProcessor() {
+        return joinMergeProcessor;
+    }
+
+    ValueJoiner<? super V, ? super V1, ? extends V2> valueJoiner() {
+        return valueJoiner;
+    }
+
+    String joinThisProcessorName() {
+        return joinThisProcessorName;
+    }
+
+    String joinOtherProcessorName() {
+        return joinOtherProcessorName;
+    }
+
+    String joinMergeProcessorName() {
+        return joinMergeProcessorName;
+    }
+
+    String thisJoinSide() {
+        return thisJoinSide;
+    }
+
+    String otherJoinSide() {
+        return otherJoinSide;
+    }
+
+    @Override
```

Comment:
These were added due to `findbugs` warnings.  It is questionable if we need `hashCode` and `equals` overrides at all. This applies to all subclasses of `StreamGraphNode`.

Reply:
Yeah, it's probably not good to *not* override them in a subclass when the superclass does override them.

I responded to your comment on the superclass, though. I think you can do without hashCode and equals in this whole class hierarchy.